### DISCUSSION TITLE
Is it possible to get rid of types.setTypeParser in PostgreSQL?

### DISCUSSION THREAD

**Discussion by lsndr** [mikro-orm/mikro-orm#4004] [Ideas]
There is a code line in the postgresql package, that makes things complicated:

1. https://github.com/mikro-orm/mikro-orm/blob/df631fd32f3b8c6901350caeb73e26a95d335077/packages/postgresql/src/PostgreSqlConnection.ts#L19

**Why does it complicate things?**

This line get called every time `getConnectionOptions()` is called, so it makes impossible to set my own date type parser. Every time I call `em.getConnection().getKnex()`, a date type parser get overriden.

**Suggestion**

1. Get rid of this line at all, since date is alway returned as string by default.
2. Move it to the beginning of the file, so it called only once.

P.S. Actually, the line 22 also has such inconvenient global behaviour, but at least it is optional.

**Comment by B4nan** [comment]
That method is/should be called only during the init phase, definitely not afterwards. I don't see how `em.getConnection()` could trigger it, can you put some logging there so find it out? `console.log(new Error().stack)` or something like that.

> Get rid of this line at all, since date is alway returned as string by default.

It's definitely not the default behaviour, if you comment that line out, some tests start to fail. Might be caused by something else (e.g. knex dialect), but the line is there on purpose, not randomly...

**Comment by lsndr** [comment]
Well, you are right. I completely forgot that in tests I used different provider, that doesn't override parsers. Now it works. Thanks
